### DISCUSSION TITLE
How to get the Maximum throughput by using the srsRAN gNB with srsUE ZMQ e2e emulator

### DISCUSSION THREAD

**Discussion by expfmmm** [srsran/srsRAN_Project#989] [General Help] [answered]
I am trying to run the srsRAN gNB with srsUE ZMQ e2e emulator with its default configuration, one gnb talk to one UE, https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html   
But I found the throughput is not maximum, for 20MHZ bandwidth, there are 106 PRBs. When I do the iperf test, only 96 RPBs are used by PDSCH during resource allocation.
![Screenshot 2024-12-16 134543](https://github.com/user-attachments/assets/531e4f0c-5373-4d9b-9bc3-c69ea14ad9fa)
How I can get the maximum throughput? thanks!

**Comment by expfmmm** [comment]
![image](https://github.com/user-attachments/assets/9bed1e9e-263c-43b3-a552-3a4638deb7dd)
I use the default gnb and ue configuration from official website. I double check the configuration yaml file, do not find an obvious parameter to limit the throughput.

**Comment by Zplusless** [comment]
Hi @expfmmm, Just wondering how to get the figure of prbs per slot you used above. Thanks in advance!

**Reply by expfmmm** [reply]
I just parse the PDSCH and PUSCH info from log file, get the PRB range, symbol range, timestamp info and slot info, then plot with python.

**Reply by Zplusless** [reply]
thanks!

**Comment by pgawlowicz** [comment]
@expfmmm how do you generate traffic?

**Reply by expfmmm** [reply]
![image](https://github.com/user-attachments/assets/63800243-3929-473a-ac36-f6ff9453458c)
![image](https://github.com/user-attachments/assets/cc3ecf92-c74c-4581-a924-d2b06460f67a)
![image](https://github.com/user-attachments/assets/7ffe3188-20cb-4123-9cd1-d1cf431caf73)

I use iperf3 to generate the traffic, the traffic is sent from core network to UE. The physical layer downlink throughput is 59Mbps and not reach to the maximum. Then I check the log file, and find 96 to 106 RB position is empty and not occupied by PDSCH.

**Reply by pgawlowicz** [reply]
could you generate UDP traffic with 100Mbps load?

**Reply by expfmmm** [reply]
![image](https://github.com/user-attachments/assets/c2228589-1bf0-414d-a841-a9f29dadad85)
![image](https://github.com/user-attachments/assets/a68d930c-e0a6-498c-bb0c-d7b4f9bfed16)
![image](https://github.com/user-attachments/assets/e78b21a0-f280-4e11-8b87-1d1afae68978)
![image](https://github.com/user-attachments/assets/045aef6e-0c8b-4c66-ad4a-cbc4e1eef8f6)
It is strange, I generate UDP traffic this time with 100Mbits/sec, the physical layer brate is still 59Mbps, PRB position from 96 to 106 is still empty.

**Reply by expfmmm** [reply, marked as answer]
I understand what happens, it is not the traffic problem, I need to configure the physical layer overhead according to 5gnr standard, and then the 96 to 106 prb can be fully occupied. Thanks!
![image](https://github.com/user-attachments/assets/2e1619dc-bc6b-4f68-9698-4b1304b20978)

**Reply by pgawlowicz** [reply]
@expfmmm  what does it mean? Which parameter do you configure?

**Reply by expfmmm** [reply]
![image](https://github.com/user-attachments/assets/2f968e6b-2a46-4682-8aa5-9eb231111349)
The DCI format 1_1 is disabled by default gnb.yml file. I do not know which DCI format gnb will select, let's say format 1_0, the frequency domain resource allocation is not via type 1. That is the main reason. The other is the coreset 1 search space should be changed to ue_dedicated.

**Reply by pgawlowicz** [reply]
@expfmmm Could you share your final gnb config?

**Comment by expfmmm** [comment]
[gnb_zmq_DCI_1_1.zip](https://github.com/user-attachments/files/18200185/gnb_zmq_DCI_1_1.zip)

**Reply by pgawlowicz** [reply]
nice, thanks @expfmmm!

**Comment by newt2105** [comment]
Hi @expfmmm, how do you generate DL traffic for UE? 
One more question: Do you try xApp to change the PRB?
Thanks!
